src/UnoFlipController.java

```diff
@@ -75,26 +75,29 @@ public void actionPerformed(ActionEvent e) {
             String[] numPlayerOptions = {"RED", "BLUE", "YELLOW", "GREEN"};
 
             JComboBox<String> comboBox = new JComboBox<>(numPlayerOptions);
-
-            int result = JOptionPane.showOptionDialog(
-                    null,
-                    comboBox,
-                    "Select colour:",
-                    JOptionPane.DEFAULT_OPTION,
-                    JOptionPane.PLAIN_MESSAGE,
-                    null,
-                    null,
-                    null);
-
-            Card.Colour colour;
-            if (result == JOptionPane.OK_OPTION) {
-                String option = (String) comboBox.getSelectedItem();
-                colour = Card.Colour.valueOf(option);
-                this.model.setCurrentColour(colour);
-                this.model.notifyViews();
-            } else {
-                System.exit(0); // absolutely must select an option, otherwise, do not start the game
+            while(true){
+                int result = JOptionPane.showOptionDialog(
+                        null,
+                        comboBox,
+                        "Select colour:",
+                        JOptionPane.DEFAULT_OPTION,
+                        JOptionPane.PLAIN_MESSAGE,
+                        null,
+                        null,
+                        null);
+
+                Card.Colour colour;
+                if (result == JOptionPane.OK_OPTION) {
```

Comment:
this block can be taken outside the loop. we only want to keep looping if the closed option is pressed. Once it is not pressed and we break out the loop, then we can look at what is inside the if-block